tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/contact/EmailAddressContact.scala

```diff
@@ -79,6 +82,26 @@ class InMemoryEmailAddressContactSearchEngine extends EmailAddressContactSearchE
     SMono.fromCallable(() => domainList.put(domain, addressContact))
       .`then`(SMono.just(addressContact))
 
+  override def delete(accountId: AccountId, mailAddress: MailAddress): Publisher[Void] =
+    getContact(accountId, mailAddress)
+      .map(contact => emailList.remove(accountId, contact))
+      .`then`()
+
+  private def getContact(accountId: AccountId, mailAddress: MailAddress): SMono[EmailAddressContact] =
```

Comment:
```suggestion
  private def getUserContact(accountId: AccountId, mailAddress: MailAddress): SMono[EmailAddressContact]
```

Or you can just overload this method ?